cmake/modules/FindMySQL.cmake

```diff
@@ -26,17 +34,22 @@ if(MYSQL_CONFIG_EXECUTABLE)
   execute_process(COMMAND ${MYSQL_CONFIG_EXECUTABLE} --libs OUTPUT_VARIABLE MYSQL_LIBRARIES OUTPUT_STRIP_TRAILING_WHITESPACE)
 else()
   find_path(MYSQL_INCLUDE_DIR mysql.h
-    /usr/local/mysql/include
-    /usr/local/include/mysql
-    /usr/local/include
-    /usr/include/mysql
-    /usr/include
-    /usr/mysql/include
-    $ENV{MYSQL_DIR}/include
-  )
+    PATH_SUFFIXES include include/mysql include/mariadb
+    HINTS ${MYSQL_DIR} $ENV{MYSQL_DIR}
+    NO_DEFAULT_PATH
+    )
+  if (NOT MYSQL_INCLUDE_DIR)
+    find_path(MYSQL_INCLUDE_DIR mysql.h
+      PATH_SUFFIXES include include/mysql include/mariadb
+      PATHS /usr/local/mysql /usr/local /usr
+      )
+  endif()
   set(MYSQL_NAMES mysqlclient mysqlclient_r)
   find_library(MYSQL_LIBRARY NAMES ${MYSQL_NAMES}
-    PATHS /usr/local/mysql/lib /usr/local/lib /usr/lib $ENV{MYSQL_DIR}/lib $ENV{MYSQL_DIR}/lib/opt
+    PATH_SUFFIXES lib lib/mariadb lib/mysql
+    lib/opt lib/opt/mariadb lib/opt/mysql
+    HINTS ${MYSQL_DIR} $ENV{MYSQL_DIR}
+    PATHS /usr/local/mysql /usr/local /usr
```

Comment:
These path suffixes and hints are broken, as they do not take into account the possibility that the library directory could be `lib32` or `lib64`. Also, `lib/opt/*` looks really strange. Why do we need to add that? Also, why add `/usr` and `/usr/local`? These are in by default already.

Reply:
This patch (mostly) just reorder the existing paths, so you are likely right that those were never needed in the first place :)

Reply:
This was the minimal change to what was already there to work for my case. It's entirely possible that further improvements need to be made for (other) new cases.